hyperemitter.js

```diff
@@ -58,13 +66,16 @@ function HyperEmitter (db, schema, opts) {
   })
 
   function process (change, enc, next) {
-    var header = headers.Event.decode(change.value)
-    var event = that.messages[header.name].decode(header.payload)
-    that._parallel(that, that._listeners[header.name] || [], event, next)
-  }
+    var container = that.codecs.Event.decode(change.value)
+    var name = container.name
+    var decoder = that.codecs[name]
+    var event = container.payload
 
-  // the status of this HyperEmitter
-  this.status = new EventEmitter()
+    if (decoder) event = decoder.decode(event)
+    else that.status.emit('no decoder found for ' + name)
```

Comment:
Instead of emitting a status message, just emit the event name with the Buffer as an argument.